### PR TITLE
fix(dashboard version service): add DashboardUID to query and responses

### LINKED ISSUE #59643: Dashboard versions: Listing versions queries ignore the dashboard UID even if it's present
**What happened**:

The [xorm](https://github.com/grafana/grafana/blob/0d4fd305c457fc1b559a67a7b283937cc899dee7/pkg/services/dashboardversion/dashverimpl/xorm_store.go#L74) and [sqlx](https://github.com/grafana/grafana/blob/0d4fd305c457fc1b559a67a7b283937cc899dee7/pkg/services/dashboardversion/dashverimpl/sqlx_store.go#L62) store implementations ignore the dashboard UID (even though it's set) when listing dashboard versions

**What you expected to happen**:

Need to give precedence to dashboard UID if it's present

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by mildwonkey**
The `DashboardUID` was not populated in the response from `Get` and `ListDashboardVersions`. This PR adds the `DashboardUID` to the Get query struct (it was already included in List) and populated the `DashboardUID `in the returned DashboardVersionDTOs from Get and List.

Addresses #59643 (I don't want to close that issue until @papagian confirms that I addressed all of the concerns in that issue)

There's one odd looking behavior I want to point out. The store methods were written such that they would succeed even if the dashboardID wasn't found (the [LEFT JOIN](https://github.com/grafana/grafana/blob/main/pkg/services/dashboardversion/dashverimpl/sqlx_store.go#L21)), so I am not returning an error if the dashboard ID or UID can't be found. This is backwards compatible, and also (reasonably) safe because the API layer already checks that the dashboard exists. I'm not opposed to changing the behavior, though; I just took my cues off the store methods.

Here's an _incredibly exiting_ comparison of the outputs from `Get /api/dashboards/id/1/versions` (truncated so it's human readable):

on main:
```
[{
	"id": 1,
	"dashboardId": 1,
	"dashboardUid": "",
         /// etc
}]
```

And on this branch:

```
[{
	"id": 1,
	"dashboardId": 1,
	"dashboardUid": "omLdn854z",
         /// etc
}]
```

Get by UID (`/api/dashboards/uid/omLdn854z/versions`) also looks good; the output is the same as above.

### CODE REVIEW

File: pkg/services/dashboardversion/dashverimpl/dashver.go [hunk 1]
```diff
@@ -69,6 +101,26 @@ func (s *Service) DeleteExpired(ctx context.Context, cmd *dashver.DeleteExpiredV
 
 // List all dashboard versions for the given dashboard ID.
 func (s *Service) List(ctx context.Context, query *dashver.ListDashboardVersionsQuery) ([]*dashver.DashboardVersionDTO, error) {
+	// Get the DashboardUID if not populated
```
Comment (ying-jeanne) [inline pkg/services/dashboardversion/dashverimpl/dashver.go:104]: nit, the two comments are identical.

File: pkg/services/dashboardversion/dashverimpl/dashver.go [hunk 2]
```diff
@@ -78,8 +130,38 @@ func (s *Service) List(ctx context.Context, query *dashver.ListDashboardVersions
...
-		dtos[i] = v.ToDTO("")
+		dtos[i] = v.ToDTO(query.DashboardUID)
 	}
 	return dtos, nil
 }
+
+// getDashUIDMaybeEmpty is a helper function which takes a dashboardID and
+// returns the UID. If the dashboard is not found, it will return an empty
+// string.
+func (s *Service) getDashUIDMaybeEmpty(ctx context.Context, id int64) (string, error) {
```
Comment (ying-jeanne) [inline pkg/services/dashboardversion/dashverimpl/dashver.go:141]: this is nice, we had a [helper](https://github.com/grafana/grafana/blob/c9773e55b234b7637ea97b671161cd856a1d3d69/pkg/api/dashboard.go#L276) which is still a method of httpserver, after your changes, we should be able to remove it.

File: pkg/services/dashboardversion/dashverimpl/dashver.go (hunk 1 above)
Comment (mildwonkey) [reply pkg/services/dashboardversion/dashverimpl/dashver.go:103]: good catch thank you !

**Review by zserge — APPROVED** [review]

**Review by papagian — APPROVED** [review]
LGTM; I have left some comment for some possible improvements

File: pkg/services/dashboardversion/dashverimpl/dashver.go [hunk 3]
```diff
@@ -14,27 +16,56 @@ const (
...
 	version, err := s.store.Get(ctx, query)
 	if err != nil {
 		return nil, err
 	}
 	version.Data.Set("id", version.DashboardID)
 
-	// FIXME: the next PR will add the dashboardService so we can grab the DashboardUID
-	return version.ToDTO(""), nil
+	// Populate the DashboardUID
+	if query.DashboardUID == "" {
```
[the remark this answers is not in the document]
Comment (mildwonkey) [reply pkg/services/dashboardversion/dashverimpl/dashver.go:61]: Yep, good call thank you! 
Fixing that brought my attention to the lines right *above* that, where we have:
```version.Data.Set("id", version.DashboardID)```

Should I be doing the same with the UID?

**Pull request merged**